### ISSUE TITLE
Bug: PCWSTR argument gets turned into PWSTR 

### ISSUE THREAD

**Issue by rkevingibson** [microsoft/windows-rs#2223]
### Which crate is this about?

windows

### Crate version

0.43.0

### Summary

In IDXCCompiler3::Compile, the parguments parameter in c++ looks like: `_In_opt_count_(argCount) LPCWSTR *pArguments,`, but gets turned into `Option<&[PWSTR]>` in the Rust. I would expect it to be mapped to `Option<&[PCWSTR]>` instead.

Ideally the use would look something like 
```
let compiler: Dxc::IDxcCompiler3 = ...
let args = [w!("arg1"), w!("arg2")];
compiler.Compile(..., Some(&args), ...);
```

Instead, something like this is required:
```
let compiler: Dxc::IDxcCompiler3 = ...
let args = [w!("arg1"), w!("arg2")];
let args = args.map(|s| PWSTR::from_raw(s.0 as *mut _));
compiler.Compile(..., Some(&args), ...);
```
which feels very unsafe, though I think it's fine as long as the strings are all constants.

### Toolchain version/configuration

Default host: x86_64-pc-windows-msvc
rustup home:  C:\Users\rkevi\.rustup

installed targets for active toolchain
--------------------------------------

wasm32-unknown-unknown
x86_64-pc-windows-msvc

active toolchain
----------------

stable-x86_64-pc-windows-msvc (default)
rustc 1.65.0 (897e37553 2022-11-02)

**Comment by kennykerr** [comment]
Good catch. Unfortunately the Win32 metadata drops the `const`-ness of the parameter, but I should be able to infer that from the fact that it is an input parameter.

**Comment by kennykerr** [comment]
Thanks for reporting. This has now been fixed (#2227) - here's an example:

https://github.com/microsoft/windows-rs/blob/3370b0d885975597b3f128ef263a03cac11f6426/crates/tests/const_ptrs/tests/test.rs#L17-L19
